File: DSA_Java/Recursion&Backtracking/Recursion with Arrays/$02displayArrayReverse.java
import java.io.*;
import java.util.*;

public class $02displayArrayReverse {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in) ;
        int n = sc.nextInt() ;
        int[] arr = new int[n] ;
        
        for(int i = 0; i < arr.length; i++){
            arr[i] = sc.nextInt() ;
        }
        
        displayArrayReverse(arr, 0) ;
    }

    public static void displayArrayReverse(int[] arr, int idx) {
        if(idx == arr.length){
            return ;    
        }
        
        displayArrayReverse(arr, idx + 1) ;
        System.out.println(arr[idx]) ;
    }

}

File: python/Recursion&Backtracking/RecursionWithArrays/$02displayArrayReverse.py
def displayArrayReverse(arr, idx) :
    if idx == len(arr) :
        return
    
    displayArrayReverse(arr, idx + 1)
    print(arr[idx])

def main() :
    n = int(input())
    arr = []
    
    for i in range(n) :
        arr.append(int(input()))
    
    displayArrayReverse(arr, 0)

main()